src/librustc_span/hygiene.rs

```diff
@@ -732,12 +733,17 @@ pub enum ExpnKind {
 }
 
 impl ExpnKind {
-    pub fn descr(&self) -> Symbol {
+    pub fn descr(&self) -> Cow<'static, str> {
```

Comment:
This is usually `String` and the function isn't supposed to be called on hot paths, so I'd rather get rid of the `Cow` ugliness.